### PR TITLE
Fixed the Java compiler tool and Java function test

### PR DESCRIPTION
**Pull request by zoejbare**
I've only seen this problem happen when running against the OpenJDK.  The proposed solution is the only one I found to work across all versions of python.

### CODE REVIEW

File: csbuild/tools/java_compilers/oracle_java_compiler.py [hunk 1]
```diff
@@ -45,11 +44,7 @@ def __init__(self, projectSettings):
 		JavaCompilerBase.__init__(self, projectSettings)
 
 		self._javaCompilerPath = os.path.join(self._javaBinPath, "javac{}".format(".exe" if platform.system() == "Windows" else ""))
-
-		try:
-			subprocess.call([self._javaCompilerPath], stdout = subprocess.PIPE, stderr = subprocess.PIPE)
-		except:
-			raise IOError("Oracle Java compiler not found at path: {}".format(self._javaCompilerPath))
+		assert os.access(self._javaCompilerPath, os.X_OK), "Oracle Java compiler not found at path: {}".format(self._javaCompilerPath)
```
Comment (ShadauxCat) [inline csbuild/tools/java_compilers/oracle_java_compiler.py:47]: I'm not sure whether or not X_OK works properly on Windows. On linux, it's literally just checking if the executable permission is set - doesn't actually confirm anything about it being able to execute SUCCESSFULLY. You can set that bit on any file. Maybe that's fine, just wanted to make sure to point it out.

Comment (ShadauxCat) [reply csbuild/tools/java_compilers/oracle_java_compiler.py:47]: Also as a sidebar - on linux, any directory you can enter will also return true for X_OK. Not sure if you care about the edge case where the path exists but is a directory. Seems unlikely to happen.

File: functional_tests/basic_java_test/tests.py [hunk 2]
```diff
@@ -70,19 +70,15 @@ def setUp(self): # pylint: disable=arguments-differ
...
-			self.assertIn("JAVA_HOME", os.environ, "JAVA_HOME must be defined in the environment if the Java binaries are not available from the system path")
-			javaExe = os.path.join(os.environ["JAVA_HOME"], "bin", javaExe)
-			self.assertFileExists(javaExe)
-
+		self.assertFileIsExecutable(javaExe)
 		self.assertMakeSucceeds("-v", "--project=hello_world", "--show-commands", "--toolchain=oracle-java")
 
 		self.assertTrue(os.access(self.outputFile, os.F_OK))
-		out = subprocess.check_output([javaExe, "-jar", self.outputFile])
+		ret, out, _ = Run([javaExe, "-jar", self.outputFile])
```
Comment (ShadauxCat) [inline functional_tests/basic_java_test/tests.py:81]: I know I'm silly... but I prefer importing modules and calling the function on the module rather than importing functions. It makes it more clear in a review that the function is not local to this file. I make exceptions for things like PlatformString and PlatformBytes because they're mimicking built-in str and bytes functions, but most things, I prefer to have the module name in the call site.

File: csbuild/tools/java_compilers/oracle_java_compiler.py (hunk 1 above)
Comment (zoejbare) [reply csbuild/tools/java_compilers/oracle_java_compiler.py:47]: I think Windows considers all files to be executable as far as that bit is concerned. I tested on Windows and it works correctly.  In this case, I think it's sufficient to just check for the existence of the file and the ability to execute it.  If it can't actually be run, that will be caught when attempting to run it.

Comment (ShadauxCat) [reply csbuild/tools/java_compilers/oracle_java_compiler.py:47]: Works for me.

**Review by ShadauxCat — APPROVED** [review]
